native/packages/react-native-bpk-component-button/src/BpkButton.android.js

```diff
@@ -76,66 +72,70 @@ const BpkButton = (props: Props) => {
     );
   }
 
-  // Validate that button is themeable and all theming attributes
-  // have been supplied. If not, disable theming.
-  if (theme) {
-    if (!isTypeThemeable(type) || !themeAttributesSupplied(type, theme)) {
-      theme = null;
-    }
+  const shouldApplyTheme =
+    themeProp &&
+    (isTypeThemeable(type) && themeAttributesSupplied(type, themeProp));
+  const theme = shouldApplyTheme ? themeProp : null;
+
+  const accessibilityTraits = ['button'];
+  const containerStyle = getStyleForElement('container', props);
+  const buttonStyle = getStyleForElement('button', props);
+  const backgroundColor = getAndroidBackgroundColour(theme, props);
+
+  if (disabled) {
+    accessibilityTraits.push('disabled');
   }
 
-  const renderIcon = (): ?Node => {
-    if (!icon) {
-      return null;
-    }
-    if (typeof icon === 'string') {
-      return (
-        <BpkIcon icon={icon} style={iconStyle(theme, props)} small={!large} />
-      );
-    }
-    return icon;
-  };
   return (
-    <BpkButtonContainer
-      accessibilityLabel={accessibilityLabel}
-      disabled={disabled}
-      icon={icon}
-      iconOnly={iconOnly}
-      onPress={onPress}
-      style={style}
-      theme={theme}
-      title={title}
-      type={type}
-      large={large}
-      {...rest}
-    >
-      <View style={getStyleForElement('view', props)}>
-        {!iconOnly && (
-          <BpkText
-            textStyle={large ? 'lg' : 'sm'}
-            emphasize
-            style={textStyle(theme, props)}
-          >
-            {Platform.OS === 'android' ? title.toUpperCase() : title}
```

Comment:
The `toUpperCase` call was accidentally removed - Android buttons should be in uppercase.

Reply:
Strange how that didn't show up in snapshots...

Reply:
I've figured out why it isn't, something we should really address if possible to prevent this:

```
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
```

Reply:
Ah I can fix this!

Reply:
see https://github.com/Skyscanner/backpack/pull/549 :)